apps/web/src/components/GalleryEditor/BetaAnnouncementModal.tsx

```diff
@@ -0,0 +1,108 @@
+import Image from 'next/image';
+import cover from 'public/beta-tester-announcement.png';
+import { useCallback } from 'react';
+import { graphql } from 'react-relay';
+import styled from 'styled-components';
+
+import { useModalActions } from '~/contexts/modal/ModalContext';
+import { BetaAnnouncementModalMutation } from '~/generated/BetaAnnouncementModalMutation.graphql';
+import { useReportError } from '~/shared/contexts/ErrorReportingContext';
+import { usePromisifiedMutation } from '~/shared/relay/usePromisifiedMutation';
+import colors from '~/shared/theme/colors';
+
+import { Button } from '../core/Button/Button';
+import { HStack, VStack } from '../core/Spacer/Stack';
+import { BaseXL, TitleL } from '../core/Text/Text';
+
+// TODO: Replace with actual typeform URL
+const TYPEFORM_URL = 'https://frameio.typeform.com/to/rgZ2Z4';
+
+export function BetaAnnouncementModal() {
+  const [optInForRoles] = usePromisifiedMutation<BetaAnnouncementModalMutation>(graphql`
+    mutation BetaAnnouncementModalMutation($input: [Role!]!) @raw_response_type {
+      optInForRoles(roles: $input) {
+        __typename
+        ... on OptInForRolesPayload {
+          user {
+            roles
+          }
+        }
+        ... on ErrInvalidInput {
+          __typename
+          message
+        }
+      }
+    }
+  `);
+
+  const { hideModal } = useModalActions();
+  const reportError = useReportError();
+
+  const handleContinue = useCallback(async () => {
+    hideModal();
+
+    try {
+      await optInForRoles({
+        variables: {
+          input: ['BETA_TESTER'],
+        },
+      });
+    } catch (error) {
+      if (error instanceof Error) {
+        reportError(error);
+      } else {
+        reportError('Uncaught error in while opting in for beta tester');
+      }
+    }
+
+    window.open(TYPEFORM_URL, '_blank');
+  }, [hideModal, optInForRoles, reportError]);
+
+  return (
+    <Container align="center">
+      <StyledImage src={cover} alt="frame" />
+
+      <StyledTextContainer align="center" gap={64}>
+        <VStack gap={32}>
+          <StyledHeader>Exclusive beta access</StyledHeader>
+          <BaseXL>
+            Thank you for being an active user on Gallery. As a thank you, we’re granting you early
+            beta access to our newest social feature — Posts!
+          </BaseXL>
+        </VStack>
+
+        <StyledButton onClick={handleContinue}>Continue</StyledButton>
+      </StyledTextContainer>
+    </Container>
+  );
+}
+
+const Container = styled(HStack)`
+  width: 1086px;
+  max-width: 100%;
+  padding: 32px 16px 20px;
+
+  @media (max-width: 1024px) {
+    flex-direction: column;
+    gap: 32px;
+  }
+`;
+
+const StyledImage = styled(Image)`
+  width: 500px;
+  height: auto;
+`;
+
+const StyledHeader = styled(TitleL)`
+  font-size: 50px;
+  color: ${colors.black.DEFAULT};
+`;
```

Comment:
Reduce the font size from `64px` to `50px` to make the title appear one line

Reply:
| Before | After |
|--------|--------|
| <img width="1211" alt="CleanShot 2023-09-07 at 15 33 50@2x" src="https://github.com/gallery-so/gallery/assets/4480258/e55983f9-d536-4bec-bc99-d42534fa75e1"> | <img width="1267" alt="CleanShot 2023-09-07 at 15 33 38@2x" src="https://github.com/gallery-so/gallery/assets/4480258/e75389f9-4fcc-4b00-933d-38fc3f763b7c"> |

Reply:
sounds good! good call